app/controllers/likes_controller.rb

```diff
@@ -0,0 +1,18 @@
+class LikesController < ApplicationController
+  def new
+    @post = Post.find(params[:post_id])
+    @like = Like.new
+  end
+
+  def create
+    @post = Post.find(params[:post_id])
+    @like = @post.likes.new(user: @user)
+
+    if @like.save
+      @post.update(likes_counter: @post.likes.count)
+      redirect_to user_post_path(@post.author, @post), notice: 'You liked the post!'
+    else
+      redirect_to user_post_path(@post.author, @post), alert: 'Unable to like the post.'
```

Comment:
- [ ] please make sure the like is created on behalf of the currently logged in user.

Reply:
- Well resolved